index.js

```diff
@@ -197,4 +197,33 @@ function popClose() {
   }
 }
 
-console.log(`${proDetail} ${popClose}`);
\ No newline at end of file
+console.log(`${proDetail} ${popClose}`);
+
+// form validation and localStorage
+
+const formEmail = document.querySelector('#email');
+const fullName = document.querySelector('#full-name');
+const Message = document.querySelector('#message');
+const form = document.querySelector('.contact-inputs');
+const errorMsg = document.querySelector('.error-message');
+
+const userData = window.localStorage.getItem('userData');
+
+const userDataFromLocal = JSON.parse(userData);
+fullName.value = userDataFromLocal.name;
+formEmail.value = userDataFromLocal.email;
+Message.value = userDataFromLocal.message;
+
+form.addEventListener('submit', (event) => {
+  if (formEmail.value !== formEmail.value.toLowerCase()) {
+    event.preventDefault();
+    errorMsg.innerHTML = `Email address should be in lower case <b>${formEmail.value.toLowerCase()}</b>`;
+  }
+  const dataObj = {
+    name: fullName.value,
+    email: formEmail.value,
+    message: Message.value,
+  };
+
+  window.localStorage.setItem('userData', JSON.stringify(dataObj));
+});
```

Comment:
- [ ] While you make a great effort working on the preservation of data in the localStorage, I noticed when a user fills the form with their data it does not populate into the localStorage. I would kindly suggest you fix this issue and be sure the data fill by the user is populated to the localStorage.

  >N.B: 1. The contact form information is persistent to page reloads.
  >2. After filling out the form, and not submitting it, the entered values should be saved in the browser.
  >3. When visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page.
  >4. If at the previous visit to the page, in the same browser, no input was entered in the form then the form inputs should remain empty.

Below is the picture of my claim 👇 

![gess local](https://user-images.githubusercontent.com/78634841/216725513-bf408d16-78e9-4c21-bee9-e650eaeb40af.png)